property/views.py

```diff
@@ -509,28 +509,23 @@ def get_queryset(self):
             (now() - datetime.timedelta(30)
              ).date(), '%Y-%m-%d'))
         date = dt.strptime(string_date, '%Y-%m-%d')
-        city = self.request.query_params['city']
+
+        city = self.request.query_params.get('city')
+
         query_results = Property.active_objects.filter(
-            address__City__icontains=city,
             last_viewed__date__gte=date,
             is_published=True,
             is_sold=False,
             view_count__gte=1).order_by('-view_count', 'last_viewed')
+        if city:
+            query_results = query_results.filter(address__City__icontains=city,
```

Comment:
@Tevinthuku the variables names `address_City__icontains` and `last_viewed__date__gte` are a little unclear on what they do. I suggest you rename them if possible and maybe consider keeping all letters as lower case in `address_City__icontains`.